app/controllers/registrations_controller.rb

```diff
@@ -746,6 +748,7 @@ def refund_payment
         payment_record.registration_payment.id,
         current_user.id,
       )
+      registration.add_history_entry({ payment_status: "refund", iso_amount: amount_left }, "user", current_user.id, 'Refund')
```

Comment:
Same as before, see whether it would make sense to merge this into the `record_refund` method directly above